client/components/home/Homepage.js

```diff
@@ -42,8 +43,15 @@ class Homepage extends React.Component {
    * @return {type}            description
    */
   onSelect(pageNumber) {
-    const offset = (pageNumber - 1) * 12;
-    this.props.displayUserDocuments(offset);
+    const searchStatus = this.props.currentState.searchParams;
+    if (searchStatus.isSearch) {
+      const offset = (pageNumber - 1) * 12;
```

Comment:
Apparently this piece of code appears a lot. You might as well move it to a function